restapi/README-docsapi.md

```diff
@@ -0,0 +1,168 @@
+* getJsonSchema
+  - gets the JSON schema for a collection.
+
+Once a schema is attached, all write operations must be done at the root of the document. In addition,
+every write will be rejected if it does not conform to the given schema.
+
+## Search Filter DSL
+
+The resources found in [ReactiveDocumentResource] can use a `where` parameter in order to filter results.
+This query parameter is valid JSON, and has data about the field to filter on, what operation to use, and the operand value.
+
+e.g. `where={"name":{"$eq":"Bob"}}` (In practice, you would probably URL-encode this)
+
+A few different features exist to match the field to something more granular:
+* Dot-syntax to represent nested paths. For a document such as `{ "a": {"b": "c"}}"`
+  you could use a where clause such as `where={"a.b": {"$eq": "c"}}`
+* Bracket syntax to reference array paths. For a document such as `{ "a": ["c"]}"`
+  you could use a where clause such as `where={"a.[0]": {"$eq": "c"}}`
+* Asterisk glob syntax to match *any* path that is not an array. For two documents such as `{ "a": {"b": "c"}}` and `{"x": {"b": "d"}}`
+  you could use a where clause such as `where={"*.b": {"$in": ["c", "d"]}}` to match both documents.
+* Array glob syntax to match *any* path that is an array. For two documents such as `{ "a": ["c"]}` and `{"a": ["a", "d"]}`
+  you could use a where clause such as `where={"a.[*]": {"$in": ["c", "d"]}}` to match both documents.
+* Comma syntax to match multiple paths. For two documents such as `{ "a": ["c"]}` and `{"x": ["c", "d"]}`
+  you could use a where clause such as `where={"a,x.[0]": {"$eq": "c"}}` to match both documents.
+* Escape sequences using `\` for literal periods, commas, and asterisks
+
+Many different filter operators can be used, each with different expected operand types and effects.
+These are split into those operations that are "CQL supported" and those that require heavier in-memory filtering
+to work properly:
+
+CQL-supported operators:
+* `$eq` - equals; expects a single text, numerical, or boolean value
+* `$lt(e)` - less than (or equal); expects a single text or numerical value (text values are compared lexically by C*)
+* `$gt(e)` - greater than (or equal); expects a single text or numerical value (text values are compared lexically by C*)
+* `$exists` - a value exists for the field; expects only the value `true` (`$exists: false` does not work as of yet due to technical limitations)
+
+CQL-unsupported operators:
+* `$in` - matching any value in a list; expects a list of values e.g. `where={"name":{"$in":["Alice", "Bob"]}}`
+* `$nin` - not matching any value in a list; expects a list of values just like $in
+* `$and` - matching two different filter operators
+* `$or` - matching at least one of many different filter operators
+
+Using a CQL-unsupported operator with a CQL-supported operator is allowed, but will potentially involve bringing large amounts of data into memory
```

Comment:
Can we also state that `$nin`, `$ne` and `$exists: false` need to evaluate for the possibly missing properties and thus need to load whole documents in memory, for `$in` we just need a single row